src/lava/lib/optimization/solvers/generic/scif/models.py

```diff
@@ -165,23 +197,88 @@ def run_spk(self) -> None:
                 a_max=2 ** 23 - 1, out=self.cnstr_intg)
 
         # !! Side effect: Changes self.beta !!
-        spk_hist_buffer = self._update_buffers()
+        # State history status:
+        # 1 -> entering ON state, 2 -> entering OFF state, 3 -> refractory
+        spk_hist_status = self._update_buffers()
+
+        local_validity, local_conflict = self._get_local_validity_conflict(
+            spk_hist_status)
 
         # Generate Sigma spikes
-        s_sig = self._gen_sig_spks(spk_hist_buffer)
+        s_sig = self._gen_sig_spks(spk_hist_status,
```

Comment:
One general question: In the Boltzmann model, each neuron sends its local contribution to the total cost to the CostIntegrator in each time step. Is that also the case for your model(s), Sumedh? Because we should change the CostIntegrator to accept only changes in the local cost, to sparsify communication.

Reply:
@phstratmann  It is indeed the case, but the sparsification should be filed as a separate issue/user story ;)